src/NuGet.Core/NuGet.ProjectModel/ProjectLockFile/LockFileDependency.cs

```diff
@@ -0,0 +1,67 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+using NuGet.Common;
+using NuGet.Packaging.Core;
+using NuGet.Shared;
+using NuGet.Versioning;
+
+namespace NuGet.ProjectModel
+{
+    public class LockFileDependency : IEquatable<LockFileDependency>
+    {
+        public string Id { get; set; }
+
+        public NuGetVersion ResolvedVersion { get; set; }
+
+        public VersionRange RequestedVersion { get; set; }
+
+        public string Sha512 { get; set; }
+
+        public PackageInstallationType Type { get; set; }
+
+        public IList<PackageDependency> Dependencies { get; set; } = new List<PackageDependency>();
+
+        public bool Equals(LockFileDependency other)
+        {
+            if (other == null)
+            {
+                return false;
+            }
+
+            if (ReferenceEquals(this, other))
+            {
+                return true;
+            }
+
+            return PathUtility.GetStringComparerBasedOnOS().Equals(Id, other.Id) &&
```

Comment:
The package id is case insensitive.

Reply:
not for linux... packageA and packagea are treated different packages

Reply:
Take my words back, it's case intensive